### PR TITLE
NXP-30393: Retrieve user info [master]

### PR DESCRIPTION
**Pull request by andreacornaglia**

### CODE REVIEW

File: nuxeo-coldstorage-web/elements/nuxeo-coldstorage-infobar.js [hunk 1]
```diff
@@ -70,12 +70,43 @@ class ColdStorageInfobar extends mixinBehaviors([FiltersBehavior, FormatBehavior
        * Input document.
        */
       document: Object,
+      /**
+       * Coldstorage status of the file.
```
Comment (RSalem07) [inline nuxeo-coldstorage-web/elements/nuxeo-coldstorage-infobar.js:74]: Can be simplified to Document's Cold Storage status

File: nuxeo-coldstorage-web/i18n/messages-fr.json [hunk 2]
```diff
@@ -22,11 +16,10 @@
   "documentContentView.restoreFromColdStorage.tooltip": "Restaurer depuis le stockage à froid",
   "documentContentView.restoreFromColdStorage.success": "Commencez à restaurer le contenu du document à partir du stockage à froid",
   "documentContentView.restoreFromColdStorage.error": "Erreur de restauration du contenu du document à partir du stockage à froid.",
-  "documentContentView.retrieveFromColdStorage.confirm" : "Confirmer",
-  "documentContentView.retrieveFromColdStorage.cancel" : "Annuler",
-  "documentContentView.retrieveFromColdStorage.ask.confirmation": "Récupérer ce document auprès du stockage à froid?",
-  "documentContentView.retrieveFromColdStorage.description": "Il en résultera un appel asynchrone et vous recevrez un courriel lorsque le document sera prêt.",
->>>>>>> NXP-30384: add manual translations for coldstorage restore feature
+  "documentContentView.retrieveFromColdStorage.confirm": "Confirmer",
```
Comment (RSalem07) [inline nuxeo-coldstorage-web/i18n/messages-fr.json:19]: You have a merge's issue on this part. `>>>>>>> NXP-30384: add manual translations for coldstorage restore feature` `<<<<<<< HEAD`

File: nuxeo-coldstorage-web/i18n/messages-pt-PT.json [hunk 3]
```diff
@@ -37,7 +25,6 @@
   "documentContentView.retrieveFromColdStorage.success": "Comece a recuperar o conteúdo do documento a partir do cold storage",
   "documentContentView.retrieveFromColdStorage.error": "Erro ao recuperar o conteúdo do documento do cold storage.",
   "documentPage.content.storedInColdStorage": "O arquivo master está arquivado no cold storage.",
->>>>>>> e99c7ad... NXP-30384: add manual translations for coldstorage restore feature
   "label.security.permission.writeColdStorage": "WriteColdStorage",
   "moveDocumentsContentsToColdStorage.ask.confirmation": "Send the documents contents to cold storage?",
```
Comment (RSalem07) [inline nuxeo-coldstorage-web/i18n/messages-pt-PT.json:29]: same than above: conflits merge  issue

File: nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/helpers/ColdStorageHelper.java [hunk 4]
```diff
@@ -290,8 +293,11 @@ public static ColdStorageContentStatus checkColdStorageContentAvailability(CoreS
                     DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), doc);
                     Instant downloadableUntil = blobStatus.getDownloadableUntil();
                     if (downloadableUntil != null) {
+                        String downloadableUntilStr = downloadableUntil.toString();
                         ctx.getProperties()
```
Comment (RSalem07) [inline nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/helpers/ColdStorageHelper.java:297]: No need to extract the variable here. Just use `downloadableUntil.toStrin()`. As reminder `toString` will give the date time in the ISO format `2011-12-03T10:15:30Z`. You should use this same date string format in the UI part.

File: nuxeo-coldstorage/src/main/resources/schemas/coldstorage.xsd [hunk 5]
```diff
@@ -8,5 +8,6 @@
   <xs:element name="coldContent" type="nxs:content" />
   <xs:element name="beingRetrieved" type="xs:boolean" />
   <xs:element name="toBeRestored" type="xs:boolean" />
+  <xs:element name="downloadableUntil" type="xs:string" />
```
Comment (RSalem07) [inline nuxeo-coldstorage/src/main/resources/schemas/coldstorage.xsd:11]: Should be a `Date`. Take a look on [Here](https://github.com/nuxeo/nuxeo-retention/blob/master/nuxeo-retention/src/main/resources/schemas/record.xsd#L19).

File: nuxeo-coldstorage-web/i18n/messages.json [hunk 6]
```diff
@@ -1,4 +1,7 @@
 {
+  "documentContentView.infobar.availableToBeRetrieved": "This is a low resolution preview. The original file can be requested by ",
```
Comment (RSalem07) [inline nuxeo-coldstorage-web/i18n/messages.json:2]: I am wondering if the message is complete `The original file can be requested by ...` something missing after `requested by `

File: nuxeo-coldstorage-web/elements/nuxeo-coldstorage-infobar.js [hunk 7]
```diff
@@ -70,12 +70,43 @@ class ColdStorageInfobar extends mixinBehaviors([FiltersBehavior, FormatBehavior
...
   }
+
+  _getDocumentStatus(document) {
+    if (document.properties['coldstorage:beingRetrieved']) {
+      return 'beingRetrieved';
+    }
+    if (document.properties['coldstorage:downloadableUntil']) {
+      return 'retrieved';
+    }
+    return 'availableToBeRetrieved';
```
Comment (RSalem07) [inline nuxeo-coldstorage-web/elements/nuxeo-coldstorage-infobar.js:101]: I am wondering if we can rename the third status `availableToBeRetrieved` as now we can do a Restore. We use for example `contentNotAvailable` or `storedInColdStorage`

File: ci/helm/preview/values.yaml [hunk 8]
```diff
@@ -43,7 +43,7 @@ nuxeo:
       pullPolicy: Always
     persistence:
       enabled: disable
-    packages: nuxeo-web-ui amazon-s3-online-storage /packages/nuxeo-coldstorage-package-*.zip
+    packages: nuxeo-web-ui-3.1.0-rc.59 amazon-s3-online-storage /packages/nuxeo-coldstorage-package-*.zip
```
Comment (RSalem07) [inline ci/helm/preview/values.yaml:46]: Could you explain why we are locking the web-ui version.

**Review by RSalem07 — COMMENTED** [review]
@andreacornaglia thx for the work I left some comments/suggestions

File: ci/helm/preview/values.yaml (hunk 8 above)
Comment (andreacornaglia) [reply ci/helm/preview/values.yaml:46]: we want to get the latest web-ui version - by default we get the last release candidate on connect pre-prod, but that doesn't have all the code we need for now.

Comment (doulba) [reply ci/helm/preview/values.yaml:46]: @andreacornaglia I think we don't have a version 3.1.0-rc.59 for WebUI.
maybe you want to use the version https://nos-preprod-connect.nuxeocloud.com/nuxeo/site/marketplace/package/nuxeo-web-ui?version=3.1.0-rc.059

Comment (andreacornaglia) [reply ci/helm/preview/values.yaml:46]: @doulba I want to use https://github.com/nuxeo/nuxeo-web-ui/tree/v3.1.0-rc.59

File: nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/helpers/ColdStorageHelper.java (hunk 4 above)
Comment (doulba) [reply nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/helpers/ColdStorageHelper.java:312]: @andreacornaglia ,
I think what Salem has said is to keep `downloadableUntil.toString()` for the mail template
and `Date.from(downloadableUntil)` only for the property value

Comment (doulba) [reply nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/helpers/ColdStorageHelper.java:312]: `
.put(COLD_STORAGE_CONTENT_AVAILABLE_UNTIL_MAIL_TEMPLATE_KEY, downloadableUntil.toString());                        doc.setPropertyValue(COLD_STORAGE_CONTENT_DOWNLOADABLE_UNITL, downloadableUntilDate);                        session.saveDocument(doc);
`

for example

File: ci/helm/preview/values.yaml (hunk 8 above)
Comment (doulba) [reply ci/helm/preview/values.yaml:46]: Ah, I think the release didn’t happen.
the latest one is 3.1.0-rc.41

Comment (andreacornaglia) [reply ci/helm/preview/values.yaml:46]: yes, I've checked with Gabriel and indeed the version is.  https://nos-preprod-connect.nuxeocloud.com/nuxeo/site/marketplace/package/nuxeo-web-ui?version=3.1.0-rc.059 - updating

File: nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/helpers/ColdStorageHelper.java (hunk 4 above)
Comment (andreacornaglia) [reply nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/helpers/ColdStorageHelper.java:312]: indeed! thanks for the catch, I'll use your suggestion!

**Review by PGoncalvesNX — APPROVED** [review]

**Review by doulba — APPROVED** [review]

**Pull request merged**